umbraco/Umbraco-CMS
# Delivery API content index handlers + unpublished/protected content handling

### Prerequisites


### Description

This PR covers two things:

- Enabling custom selectors, filters and sorts to add data to the delivery API content index
- Explicitly filter out unpublished and protected content

_Note: Published variants handling is slated for another task._

### Testing

1. Rebuild the delivery API content index.
2. Verify that the delivery API querying part work as per usual.
3. Create some unpublished and some protected content structures.
4. Rebuild the index.
5. Verify that neither the unpublished nor the protected content is present in the index.
